odoo/addons/base/tests/test_tests_tags.py

```diff
@@ -0,0 +1,234 @@
+# -*- coding: utf-8 -*-
+# Part of Odoo. See LICENSE file for full copyright and licensing details.
+
+import unittest
+
+from odoo.tests.common import tagged, TagsTestSelector, TagsError
+
+
+@tagged('nodatabase')
+class TestSetTags(unittest.TestCase):
+
+    def test_set_tags_empty(self):
+        """Test the set_tags decorator with an empty set of tags"""
+
+        @tagged()
+        class FakeClass():
+            pass
+
+        fc = FakeClass()
+
+        self.assertTrue(hasattr(fc, 'test_tags'))
```

Comment:
I am just wondering: in the actual test selection, do we test the *class* or an *instance*?
If it is the class, the test should apply on the class as well.

Reply:
AFAIK we test the instance:
https://github.com/odoo/odoo/blob/c6a3f2b3b66502e03e493cff6c301b3c38c74306/odoo/modules/module.py#L536

Reply:
Thanks for clarifying :+1: